Framework/Algorithms/test/CorelliPowderCalibrationApplyTest.h

```diff
@@ -0,0 +1,162 @@
+    CorelliPowderCalibrationApply alg;
+    alg.initialize();
+    alg.setPropertyValue("Workspace", "wrongTypeWs");
+    alg.setPropertyValue("CalibrationTable", calTableName);
+
+    // ensure that an exception is thrown here
+    TS_ASSERT_THROWS(alg.execute(), const std::runtime_error &);
+  }
+
+  void testValidateCalTable() {
+    // get a mock workspace with correct instrument name
+    auto ws = createTestEventWorkspace();
+
+    // get a mock calTable with wrong header
+    std::string calTableName = "wrongCalTable";
+    auto calTable = createTestCalibrationTableWorkspace(calTableName);
+    calTable->removeColumn("Xposition");
+
+    // setup alg
+    CorelliPowderCalibrationApply alg;
+    alg.initialize();
+    alg.setPropertyValue("Workspace", "correctWs");
+    alg.setPropertyValue("CalibrationTable", calTableName);
+
+    // ensure that an exception is thrown here
+    TS_ASSERT_THROWS(alg.execute(), const std::runtime_error &);
+  }
+
+  void testExec() {
+    // setup input workspace
+    auto ws = createTestEventWorkspace();
+
+    // get a correct mock calibration table
+    std::string calTableName = "correctCalTable";
+    auto calTable = createTestCalibrationTableWorkspace(calTableName);
+
+    // setup alg
+    CorelliPowderCalibrationApply alg;
+    alg.initialize();
+    alg.setPropertyValue("Workspace", "correctWs");
+    alg.setPropertyValue("CalibrationTable", calTableName);
+
+    // make sure no exception is thrown here
+    TS_ASSERT_THROWS_NOTHING(alg.execute())
+  }
+
+private:
+  EventWorkspace_sptr createTestEventWorkspace() {
+    // Name of the output workspace.
+    std::string outWSName("correctWs");
+
+    IAlgorithm_sptr lei =
+        AlgorithmFactory::Instance().create("LoadEmptyInstrument", 1);
+    lei->initialize();
+    lei->setPropertyValue("Filename", "CORELLI_Definition.xml");
+    lei->setPropertyValue("OutputWorkspace", outWSName);
+    lei->setPropertyValue("MakeEventWorkspace", "1");
+    lei->execute();
+
+    EventWorkspace_sptr ws;
+    ws = AnalysisDataService::Instance().retrieveWS<EventWorkspace>(outWSName);
+    return ws;
+  }
+
+  TableWorkspace_sptr
+  createTestCalibrationTableWorkspace(const std::string &outWSName) {
+
+    ITableWorkspace_sptr itablews = WorkspaceFactory::Instance().createTable();
+    AnalysisDataService::Instance().addOrReplace(outWSName, itablews);
+
+    TableWorkspace_sptr tablews =
+        std::dynamic_pointer_cast<TableWorkspace>(itablews);
+    TS_ASSERT(tablews);
+
+    // Set up columns
+    for (size_t i = 0;
+         i < CorelliCalibration::calibrationTableColumnNames.size(); i++) {
+      std::string colname = CorelliCalibration::calibrationTableColumnNames[i];
+      std::string type = CorelliCalibration::calibrationTableColumnTypes[i];
+      tablews->addColumn(type, colname);
+    }
+
+    // append rows
+    Mantid::API::TableRow sourceRow = tablews->appendRow();
+    sourceRow << "source" << 0. << 0. << -15.560 << 0. << 0. << 0. << 0.;
```

Comment:
Just to be consistent, would you use CORELLI's component names "moderator" and "sample-position"?

Reply:
yep, the components are updated.